clients/template/template.go

```diff
@@ -81,7 +81,7 @@ func (c Client) Validate(ctx context.Context, params *ValidateParams) error {
 	}
 
 	if apiErr, ok := err.(api.GenericOpenAPIError); ok {
-		if summary, ok := apiErr.Model().(*api.TemplateSummary); ok {
+		if summary, ok := apiErr.Model().(*api.TemplateSummaryError); ok {
```

Comment:
Any way that this could do less type assertion? For example, does apiErr.Error() now return something more reasonable, or could we make that happen?

Reply:
With type assertion to `TemplateSummaryError`:
```
Error: template failed validation:
        kind=unknown field= reason="invalid kind"
```
With type assertion to `GenericOpenAPIError`:
```
Error: template failed validation:
        kind=unknown field= reason="invalid kind"
```
Without type assertion:
```
Error: failed to validate template: kind=unknown field= reason="invalid kind"
```

Seems like we can get away with just asserting to `GenericOpenAPIError` now